subsys/bluetooth/host/scan.c

```diff
@@ -935,6 +935,14 @@ void bt_hci_le_per_adv_sync_established(struct net_buf *buf)
 	memset(&sync_info, 0, sizeof(sync_info));
 	sync_info.interval = pending_per_adv_sync->interval;
 	sync_info.phy = bt_get_phy(pending_per_adv_sync->phy);
+
+	/* When the periodic advertiser list is used, or address resolution is
+	 * enabled, the address and SID may not be identical to those passed in
+	 * by the application.
+	 */
+	bt_addr_le_copy(&pending_per_adv_sync->addr, &evt->adv_addr);
+	pending_per_adv_sync->sid = evt->sid;
```

Comment:
I can't help but to wonder if we should perhaps keep and include both addresses, like we do for `bt_conn`, where we have both the application provided address, as well as the RPA or resolved address. 

Does the SID really also change?

Reply:
From the HCI spec: 
> If the Periodic Advertiser List is not used, the Advertising_SID, Advertiser Address_Type, and Advertiser Address
parameters specify the periodic advertising device to listen to; otherwise they shall be ignored

So if the list is used, then the values should probably not have been set to the anything, and this change is absolutely fine. 

Still looking for the case where address resolution is enabled.

Reply:
Although I would suggest to check the `BT_PER_ADV_SYNC_SYNCING_USE_LIST` option before copying the address, as there's no reason to do that if `BT_PER_ADV_SYNC_SYNCING_USE_LIST` isn't set (small optimization).

Reply:
I believe this may happen:
* Peer address is set to a resolvable address
* Address resolution is enabled, the scanner is able to resolve the address.
* The scanner may sync to this device and report and put the resolved address in the Sync Established event.

I'm sure some implementations just check the resolved address in this case as that makes most sense. In any case, this implementation would cover that case as well.

Reply:
Regarding checking the `BT_PER_ADV_SYNC_SYNCING_USE_LIST` I'm fine with adding such a check, but that will increase code size without reason

Reply:
Looks like we can also guard with `bt_dev.le.rl_entries` which determines if address resolution is enabled or not. 

But in general, I'd like to avoid having an API that takes Address_A and when returns Address_A' (without Address_A also included). Address_A' should be an addition, and not a replacement IMO (similar to the `conn.h` API).

Reply:
> I believe this may happen:
> 
>     * Peer address is set to a resolvable address
> 
>     * Address resolution is enabled, the scanner is able to resolve the address.
> 
>     * The scanner may sync to this device and report and put the resolved address in the Sync Established event.
> 
> 
> I'm sure some implementations just check the resolved address in this case as that makes most sense. In any case, this implementation would cover that case as well.

But it may also happen that

- Peer address is set to a resolvable address
- Address cannot yet be resolved due to missing IRK
- Sync establish is requested by application the RPA
- We obtain the IRK and can resolve the adresss
- Sync established is resolved, but where the event has the resolved address, and the RPA that the application provided

Which is exactly what this PR attempts to fix, right? Except that now the application will see a different address than what it provided in `bt_le_per_adv_sync_create`, which I would like to avoid. This can be avoided by adding another address field in the `bt_le_per_adv_sync` struct and `bt_le_per_adv_sync_synced_info` struct, so that the application will get the same address it requested, as well as information about whether the address has since then been resolved.

Reply:
> Regarding checking the `BT_PER_ADV_SYNC_SYNCING_USE_LIST` I'm fine with adding such a check, but that will increase code size without reason

It isn't without reason if we avoid multiple unnecessary assignments and a memcpy in my opinion :)